src/main/java/com/google/roomies/database/Database.java

```diff
@@ -0,0 +1,76 @@
+package com.google.roomies;
+
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map; 
+
+interface Database {
+  /**
+  * Gets the database instance.
+  *
+  * @throws IOException if database cannot be initialized
+  */
+  public Object getDatabase() throws IOException;
+
+  /**
+  * Sets the database instance.
+  * Used to set database to a mock for testing.
+  *
+  * @throws IOException if database cannot be initialized
+  */
+  public void setDatabaseForTest(Firestore database);
+  
+  /**
+  * Add a document to a collection using a map.
+  *
+  * @param collectionName name of collection in database
+  * @param doc document that implements the document interface
+  */  
+  public void addDocumentAsMap(String collectionName, Document doc);
```

Comment:
General thought: I'm thinking an API which ties closer to your data model would be easier to work with. For instance, instead of having all these API calls which work with a map, what if you have the consume/return the autovalues directly? Then you would have insertLeaseListing(), getLeaseListing(), insertComment(), etc

One advantage of this route is that it limits the chance you end up with bad stuff in your database - then you only have to worry about getting the Object type correct in the toMap method of each autovalue.